### DISCUSSION TITLE
Serde support

### DISCUSSION THREAD

**Discussion by solaeus** [tree-sitter/tree-sitter#4694] [Ideas]
Is it possible to implement serde's Serialize and Deserialize traits for types like Parser and Tree? I would like to see it done and can easily do it myself but I don't know how much of a barrier the FFI creates or if there is a conceptual reason it would not work.

**Comment by ObserverOfTime** [comment]
See https://github.com/tree-sitter/py-tree-sitter/issues/98#issuecomment-1084796290. But this is useful so perhaps serialising trees/nodes in a binary and compressed form (e.g. CBOR+zlib) would be fine. It probably should not be part of the core library though.
